module/VuFindSearch/src/VuFindSearch/Backend/Solr/Command/WriteDocumentCommand.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Command to write a document object to Solr.
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2021.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+namespace VuFindSearch\Backend\Solr\Command;
+
+use VuFindSearch\Backend\BackendInterface;
+use VuFindSearch\Backend\Solr\Connector;
+use VuFindSearch\Backend\Solr\Document\DocumentInterface;
+use VuFindSearch\Command\CommandInterface;
+use VuFindSearch\ParamBag;
+
+/**
+ * Command to write a document object to Solr.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class WriteDocumentCommand extends \VuFindSearch\Command\AbstractBase
+{
+    /**
+     * Constructor.
+     *
+     * @param string            $backend Search backend identifier
+     * @param DocumentInterface $doc     Document to write
+     * @param ?int              $timeout Timeout value (null for default)
+     * @param string            $handler Handler to use
+     * @param ?ParamBag         $params  Search backend parameters
+     */
+    public function __construct(
+        string $backend,
+        DocumentInterface $doc,
+        int $timeout = null,
+        string $handler = 'update',
+        ?ParamBag $params = null
+    ) {
+        parent::__construct($backend, compact('doc', 'timeout', 'handler'), $params);
+    }
+
+    /**
+     * Execute command on backend.
+     *
+     * @param BackendInterface $backend Backend
+     *
+     * @return CommandInterface Command instance for method chaining
+     */
+    public function execute(BackendInterface $backend): CommandInterface
+    {
+        $this->validateBackend($backend);
+        $connector = is_callable([$backend, 'getConnector'])
+            ? $backend->getConnector() : null;
```

Comment:
Although it would not be as compact and efficient as this code here, I wonder if it would make sense to have an `AbstractBase::validateMethod()` or similar as well? But this is not something that needs to be done before 8.0.

Reply:
What signature/behavior did you have in mind? I'm not sure if I follow exactly what you have in mind.

Reply:
I don't think I had anything specific in mind 😄, but if you are refactoring this in any case then we can just forget about this...

Reply:
Good point!